### ISSUE TITLE
make install does not compile gap_fit even though HAVE_GAP=1

### ISSUE THREAD

**Issue by stenczelt** [libAtoms/QUIP#293]
Even though I have `HAVE_GAP=1` in the config, running `make install` does not compile `gap_fit`, nor is the teach_sparse dummy script copied over, so I think something is not being triggered in the Makefile.

running `make gap_programs` and then `make install` does work though.

**Comment by noambernstein** [comment]
On Thu, May 6, 2021 at 8:23 AM Tamas Stenczel ***@***.***>
wrote:

> Even though I have HAVE_GAP=1 in the config, running make install does
> not compile gap_fit, nor is the teach_sparse dummy script copied over, so
> I think something is not being triggered in the Makefile.
>

There _used to be_ a separate make var for fitting.  I just checked Makefile.config, and it's not there.  Maybe it was removed when GAP was completely refactored?

**Comment by bernstei** [comment]
Aha - top Makefile "install" target depends on PROGRAMS, but that does not include GAP_PROGRAMS.  "all" target depends explicitly on PROGRAMS and GAP_PROGRAMS.  I always do "make all" before the install, but if you don't, I can see how it'd fail.  Either the "install" target has to depend on GAP_PROGRAMS, or we need `PROGRAMS += GAP_PROGRAMS`, I think. (@stenczelt I think we are some of the few who use "make install" - lots of others just run out of the build directory).

**Comment by stenczelt** [comment]
Yeah, makes sense. I think if GAP is present then programs should include gap related programs as well.
